packages/flight-icons/scripts/build-parts/generateBundleSVGJS.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright IBM Corp. 2021, 2025
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import fs from 'fs-extra';
+import cheerio from 'cheerio';
+import prettier from 'prettier';
+import path from 'path';
+
+import { ConfigData } from '../@types/ConfigData';
+import { AssetsCatalog } from '../@types/AssetsCatalog';
+
+const prettierConfig = {
+    parser: 'typescript',
+    tabWidth: 4,
+    singleQuote: true,
+    trailingComma: 'none'
+} as const;
+
+const getSymbolFunction = (sourceSvg: string): string => {
+    const $ = cheerio.load(sourceSvg, { xmlMode: true });
+    const $svg = $('svg');
+    const viewBox = $svg.attr('viewBox') || '0 0 16 16';
+
+    $('[fill]').each((_i: number, el: unknown) => {
+        const val = $(el).attr('fill');
+        if (val && val !== 'none') {
+            $(el).attr('fill', 'currentColor');
+        }
+    });
```

Comment:
I think you can keep things much simpler, using a simple regex on the `innerContent` string (see how it's done here, for example: https://github.com/hashicorp/design-system/blob/main/packages/flight-icons/scripts/build-parts/generateBundleSVGSprite.ts#L43-L48)